### ISSUE TITLE
Themes are busted

### ISSUE THREAD

**Issue by nickdrozd** [Wilfred/remacs#744]
1. Start Remacs with `-Q` and do `eval-expression`: `(load-theme 'light-blue)`. For me, this causes the whole window to fill up with light blue, after which Remacs freezes. Other themes freeze up too, but only some. `(load-theme 'wheatgrass)`, for instance, doesn't seem to cause any problems. (You can run `customize-themes` and play around with the option presented to see which ones work and which ones don't.)

2. Start Remacs with `-Q` and do `eval-expression`: `(load-theme 'manoj-dark)`. Now do `eval-expression` and start typing `(load-theme`. For me, Remacs freezes at this point, apparently when it tries to load the function signature. Sometimes if I wait long enough I can `C-g` and regain control, but usually it just remains frozen. I use `manoj-dark`, so this is a problem for me.

`Remacs 26.0.60 (build 8, x86_64-apple-darwin16.7.0, NS appkit-1504.83 Version 10.12.6 (Build 16G29)) of 2018-05-14`

**Comment by cjohansson** [comment]
I can reproduce the error and it seems identical to [re-builder freezes buffers](https://github.com/Wilfred/remacs/issues/717). I bet it has something to do with `redisplay`

If I start remacs with `-Q -nw` then issue cannot be reproduced. Do you use macOS as well?

**Comment by cjohansson** [comment]
Another strange phenomena is if you start the mac application, issue cannot be reproduced. Like this:

`open nextstep/Remacs.app`

**Comment by shaleh** [comment]
```
#!/bin/bash

echo "Testing build and compile of remacs with arguments: $@"

rust_src/admin/build-and-run-remacs.sh "$@"

result=$?

if [ $? -ne 0 ]; then
    exit $result
fi

read -r -p "Did it work? [Ny]" work
case "$work" in
    Y|y)
        exit 0
        ;;
    *)
        exit 1
        ;;
esac
```

**Comment by shaleh** [comment]
Also, this bug happens on Linux too.

**Comment by shaleh** [comment]
2d47adc4c431081a6b252ab13d63b39cd36750b3 is the offending commit.

**Comment by nickdrozd** [comment]
@shaleh This is still busted for me.

**Comment by shaleh** [comment]
@nickdrozd indeed. for light-blue I still see the freeze. manoj-dark and my solarized theme work though. Interesting. I am re-opening this.

**Comment by shaleh** [comment]
Please try again with current master.

**Comment by cjohansson** [comment]
Works for me on macOS 10.13.5 👍
